src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -194,7 +194,7 @@ private static InstanceDiscoveryMetadataEntry CreateEntry(Uri orginalAuthority,
         {
             return new InstanceDiscoveryMetadataEntry()
             {
-                Aliases = new[] { regionalizedAuthority.Host },
+                Aliases = new[] { regionalizedAuthority.Host, orginalAuthority.Host },
                 PreferredCache = orginalAuthority.Host,
```

Comment:
i don't think this is needed once the actual issue on line 261 is fixed.

Reply:
Do we think that this is the right fix? If we add the `originalAuthorityHost` to this new `InstanceDiscoveryMetadataEntry` we'll have several  `InstanceDiscoveryMetadataEntry` which `Aliases` property will contain the same value, therefore the  `InstanceDiscoveryMetadataEntry` won't be a partition any longer?
Wouldn't this bring some random behavior?

Reply:
@neha-bhargava to look at this.

Reply:
I had this earlier, the alias with the original authority, removed it because then it will cache the instance discovery with the original alias and if regional is set to false for some percentage, then still it will go to the regional endpoint. So, I don't think this is needed.

Reply:
Reverted this after discussion. Not the right fix.

Reply:
what's the right fix then?

Reply:
Not sure, more investigation is needed. Will create a separate issue for this.

Reply:
have a link to the issue?

---
In reply to: [572444297](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2393#discussion_r572444297) [](ancestors = 572444297)